src/streams/EditStreamCard.js

```diff
@@ -5,45 +5,96 @@ import { View } from 'react-native';
 
 import Input from '../common/Input';
 import ZulipTextIntl from '../common/ZulipTextIntl';
-import SwitchRow from '../common/SwitchRow';
 import ZulipButton from '../common/ZulipButton';
-import styles, { createStyleSheet } from '../styles';
-import { IconPrivate } from '../common/Icons';
+import styles from '../styles';
+import SelectableOptionRow from '../common/SelectableOptionRow';
 
-const componentStyles = createStyleSheet({
-  switchRow: {
-    paddingLeft: 8,
-    paddingRight: 8,
-  },
-});
+type AccessPolicy =
+  | 'public'
+  | 'web_public'
+  | 'private_shared_history'
+  | 'private_protected_history';
+
+type StreamPolicySettings = {
+  // The properties on a stream object that are related to the access policy.
+  isPrivate: boolean,
+  isWebPublic: boolean,
+  historyPublicToSubscribers: boolean,
+};
 
 type Props = $ReadOnly<{|
   isNewStream: boolean,
   initialValues: {|
     name: string,
     description: string,
     invite_only: boolean,
+    is_web_public: boolean | void,
+    history_public_to_subscribers: boolean,
   |},
-  onComplete: (name: string, description: string, isPrivate: boolean) => void,
+  onComplete: (name: string, description: string, policySettings: StreamPolicySettings) => void,
 |}>;
 
 type State = {|
   name: string,
   description: string,
-  isPrivate: boolean,
+  accessPolicy: AccessPolicy,
 |};
 
+const accessPolicyFromStreamSettings = (streamSettings: StreamPolicySettings): AccessPolicy => {
+  const { isWebPublic, isPrivate, historyPublicToSubscribers } = streamSettings;
+  if (isWebPublic) {
+    return 'web_public';
+  } else if (!isPrivate) {
+    return 'public';
+  } else if (isPrivate && historyPublicToSubscribers) {
+    return 'private_shared_history';
+  } else {
+    return 'private_protected_history';
+  }
+};
+
+const streamAccessPolicies = {
+  web_public: {
+    title: 'Web-public',
+    policy: { isPrivate: false, isWebPublic: true, historyPublicToSubscribers: true },
+    description:
```

Comment:
I've put these descriptions here because they are present in the webapp and 
1. We might want to display them in some UI redesign in the near future.
2. They give the reader of this code a real description of what each policy means.
They are however, unused. So I'm not against removing them.

Reply:
Seems fine.

Let's include a comment noting that they aren't currently used, and therefore aren't in the set of strings to be translated. That way we'll know to add them to be translated if we start using them in the future.